src/test/kotlin/org/rust/lang/core/macros/decl/RsMacroExpansionTest.kt

```diff
@@ -1128,6 +1158,25 @@ class RsMacroExpansionTest : RsMacroExpansionTestBase() {
         (2 + 2)
     """)
 
+    fun `test matches! macro`() = checkSingleMacro("""
+        macro_rules! matches {
+            ($ expression:expr, $( $ pattern:pat_param )|+ $( if $ guard: expr )? $(,)?) => {
+                match $ expression {
+                    $( $ pattern )|+ $( if $ guard )? => true,
+                    _ => false
+                }
+            }
+        }
+        fn main() {
+            let _ = matches!(Some(1), Some(2) | Some(3));
```

Comment:
Would this feature also work with the newly introduced [Or patterns](https://blog.rust-lang.org/2021/06/17/Rust-1.53.0.html#or-patterns)?

```rust
let _ = matches!(Some(1), Some(2 | 3));
```

Reply:
Yep
![image](https://user-images.githubusercontent.com/2539310/122676612-75dcfe80-d1e7-11eb-8cb9-17ac70e5cb0d.png)

Reply:
@vlad20012 let's add a test with or_patterns